File: apps/angular-web/libs/purchasing/data-access/src/lib/models/purchasing-analytics.model.ts
/** A single vendor's total spend and its running cumulative share of total spend, for the Pareto view. */
export interface VendorSpendDto {
  readonly vendorId: number;
  readonly vendorName: string;
  readonly totalSpend: number;
  /** Running share of total spend as a percentage; monotonically increasing, the last entry is 100. */
  readonly cumulativePercent: number;
}

/** Purchase order count and value rolled up by status, for the pipeline summary. */
export interface PipelineSummaryItemDto {
  /** Human-readable status: Pending, Approved, Rejected, or Complete. */
  readonly statusLabel: string;
  readonly poCount: number;
  readonly totalValue: number;
}

/** Aggregate purchasing analytics (`GET /v1/purchasing/analytics`) — Pareto vendor spend plus the PO pipeline summary. */
export interface PurchasingAnalyticsDto {
  /** Every vendor, ordered by total spend descending. */
  readonly paretoData: VendorSpendDto[];
  /** Always exactly four entries — Pending, Approved, Rejected, Complete — in that order. */
  readonly pipelineSummary: PipelineSummaryItemDto[];
}

File: apps/api-dotnet/src/AdventureWorks.Models/Features/Purchasing/PurchasingAnalyticsModel.cs
namespace AdventureWorks.Models.Features.Purchasing;

/// <summary>
/// Aggregate purchasing analytics (<c>GET /api/v1/purchasing/analytics</c>): Pareto vendor-spend
/// data plus a purchase order pipeline summary.
/// </summary>
public sealed class PurchasingAnalyticsModel
{
    /// <summary>
    /// Every vendor, ordered by total spend descending (ties broken by <c>VendorId</c> ascending),
    /// with a running cumulative percentage of total spend. Vendors with no purchase orders are
    /// included with a total spend of zero.
    /// </summary>
    public List<VendorSpendModel> ParetoData { get; set; } = [];

    /// <summary>
    /// Purchase order counts and value by status, in status-code order (Pending, Approved,
    /// Rejected, Complete). All four statuses are always present, even when a status has no
    /// purchase orders.
    /// </summary>
    public List<PurchaseOrderPipelineStatusModel> PipelineSummary { get; set; } = [];
}
